### PR TITLE
Replace LightClient single_step JSON tests with MBT

### LINKED ISSUE #640: Replace all LightClient single_step JSON tests with MBT
As a step towards full replacement of the current LC testing with model-based tests, we are going to replace all [single step JSON fixtures](https://github.com/informalsystems/tendermint-rs/tree/master/light-client/tests/support/single_step) with a combination of:
 * MBT tests from TLA+ assertions
 * mutations on top of those.

The result of this issue should be the complete removal of the above directory, while preserving the test coverage.

### PR DESCRIPTION
**Pull request by andrey-kuprianov**
closes #640 

* [ ] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG.md

### CODE REVIEW

**Comment by andrey-kuprianov** [comment]
Besides clippy warnings, looks good from my side:) Thanks @Shivani912 for finishing this!

File: light-client/tests/model_based.rs [hunk 1]
```diff
@@ -440,7 +443,7 @@ fn single_step_test(
         tc.initial.signed_header.clone(),
         tc.initial.next_validator_set.clone(),
     );
-    let clock_drift = Duration::from_secs(1);
+    let clock_drift = Duration::from_secs(0);
```
Comment (romac) [inline light-client/tests/model_based.rs:446]: Just wondering what's the purpose of this change? Does it strengthen the tests somehow by making them not rely on any compensation for the clock drift?

**Review by romac — APPROVED** [review]

**Comment by romac** [comment]
Also, could you point me at the changes in the TLA+ specs/models that accounts for the change in https://github.com/informalsystems/tendermint-rs/pull/657?

**Comment by andrey-kuprianov** [comment]
> Also, could someone point me at the changes in the TLA+ specs/models that accounts for the change in #657?

@romac it is described in #659. We have disabled the tests that were checking this for now; will reenable them when the TLA+ model is updated.

**Comment by romac** [comment]
Thanks!

File: light-client/tests/model_based.rs (hunk 1 above)
Comment (andrey-kuprianov) [reply light-client/tests/model_based.rs:446]: We have disabled the clock drift, because the TLA+ model doesn't account for it at the moment. As a result, some tests were failing non-deterministically. As soon as the clock drift is introduced into the TLA+ model (@konnov and @josef-widder are working on that), we will also enable it here,

Comment (romac) [reply light-client/tests/model_based.rs:446]: Thanks for the clarification!

**Comment by romac** [comment]
@Shivani912 @andrey-kuprianov Great job!

File: light-client/tests/model_based.rs (hunk 1 above)
Comment (Shivani912) [reply light-client/tests/model_based.rs:446]: The model does not account for `clock_drift` and so several timing-related tests were failing where the difference was of 1 second between a "valid" and "invalid" header. So it just made sense to remove this param from the test as well

**Comment by Shivani912** [comment]
Thanks @romac !

File: light-client/tests/model_based.rs (hunk 1 above)
Comment (konnov) [reply light-client/tests/model_based.rs:446]: This PR introduces clock drift in the TLA+ spec: https://github.com/tendermint/spec/pull/216

**Comment by andrey-kuprianov** [comment]
@romac thank you for reviewing the PRs so quickly:)
